iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -54,16 +63,44 @@ public ShuntCompensatorAdder setCurrentSectionCount(int currentSectionCount) {
         return this;
     }
 
+    @Override
+    public ShuntCompensatorAdder setRegulatingTerminal(Terminal regulatingTerminal) {
+        this.regulatingTerminal = (TerminalExt) regulatingTerminal;
+        return this;
+    }
+
+    @Override
+    public ShuntCompensatorAdder setVoltageRegulatorOn(boolean voltageRegulatorOn) {
+        this.voltageRegulatorOn = voltageRegulatorOn;
+        return this;
+    }
+
+    @Override
+    public ShuntCompensatorAdder setTargetV(double targetV) {
+        this.targetV = targetV;
+        return this;
+    }
+
+    @Override
+    public ShuntCompensatorAdder setTargetDeadband(double targetDeadband) {
+        this.targetDeadband = targetDeadband;
+        return this;
+    }
+
     @Override
     public ShuntCompensatorImpl add() {
         String id = checkAndGetUniqueId();
         TerminalExt terminal = checkAndGetTerminal();
         ValidationUtil.checkbPerSection(this, bPerSection);
         ValidationUtil.checkSections(this, currentSectionCount, maximumSectionCount);
+        ValidationUtil.checkRegulatingTerminal(this, regulatingTerminal, getNetwork());
+        ValidationUtil.checkVoltageControl(this, voltageRegulatorOn, targetV);
+        ValidationUtil.checkTargetDeadband(this, "shunt compensator", targetDeadband);
         ShuntCompensatorImpl shunt
                 = new ShuntCompensatorImpl(getNetwork().getRef(),
-                                           id, getName(), bPerSection, maximumSectionCount,
-                                           currentSectionCount);
+                id, getName(), bPerSection, maximumSectionCount,
+                currentSectionCount, regulatingTerminal == null ? terminal : regulatingTerminal,
```

Comment:
We need to document that if the regulating terminal is null, then the regulation becomes local. Is it the same assumption for SVC, generators, tap changers ?

Reply:
It is the same for generators and SVC but not for tap changers.

Reply:
Maybe we should fix the bahavior for tap changers to be consistent everywhere. At least we should create an issue to report that.